### ISSUE TITLE
Malformed path for Qt5::Core in build using macOS homebrew. Update build instructions?

### ISSUE THREAD

**Issue by rakenodiax** [leela-zero/leela-zero#2177]
Hello,

With the `next` branch checked out, I'm following the build instructions for macOS, and CMake is failing to build with the following error:

```
CMake Error at /usr/local/lib/cmake/Qt5Core/Qt5CoreConfig.cmake:15 (message):
  The imported target "Qt5::Core" references the file

     "/usr/local/.//mkspecs/macx-clang"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "/usr/local/lib/cmake/Qt5Core/Qt5CoreConfigExtras.cmake"

  but not all the files it references.

Call Stack (most recent call first):
  /usr/local/lib/cmake/Qt5Core/Qt5CoreConfigExtras.cmake:50 (_qt5_Core_check_file_exists)
  /usr/local/lib/cmake/Qt5Core/Qt5CoreConfig.cmake:169 (include)
  CMakeLists.txt:50 (find_package)
```

I'm running macOS High Sierra version 10.13.6 (build 17G4015)

**Comment by gcp** [comment]
This is a macOS homebrew bug, it doesn't add the qt5 binaries to the path correctly.

Can be worked around by manually adding /usr/local/opt/qt5/bin to the PATH.

**Comment by gcp** [comment]
Closing as AFAIK this can't be fixed on our side.

**Comment by rakenodiax** [comment]
@gcp hmm... manually adding the qt5 bin path to my PATH did not work. Upon further inspection, it looks like it might be a bug in Qt5 itself, as it is referencing the malformed path.

Thanks!

**Comment by rakenodiax** [comment]
That worked for me, thanks! Would it be possible to add this to the macOS build instructions, as Homebrew is probably how users will be installing QT5

**Comment by wonderingabout** [comment]
@Hersmunch 

thanks for remembering

i dont use mac myself (i'm on linux), but i had already wrotten these clearer instructions quite some time ago : https://github.com/leela-zero/leela-zero/pull/2071

i added specific mac instructions based on the feedback of mac users, see the PR discussion (i documented the links of the discussion and summarized it)

**Comment by gcp** [comment]
>That worked for me, thanks! Would it be possible to add this to the macOS build instructions, as Homebrew is probably how users will be installing QT5

Should probably try to find the bug filed against Homebrew and link to that.

I'd rather not have done this but this issue has been there for so long it looks like they're unable to fix it.

**Comment by wonderingabout** [comment]
so, what do we do with the mac instructions ?

put them back to default and ignore the mac users feedback ?

personally both are fine to me because i dont use mac, but if it really helps to add these mac specific instructions i'd like to add them even if we remove them later on

**Comment by rakenodiax** [comment]
I think it makes sense that if a specific package manager is already mentioned in the build instructions, then other steps which are necessary when using the same should also be included. For example, these instructions would help new users from encountering the same problem I did:

```sh
# Clone github repo
git clone https://github.com/gcp/leela-zero
cd leela-zero
git submodule update --init --recursive

# Install build depedencies using homebrew
brew install boost cmake qt5

# Use stand alone directory to keep source dir clean
mkdir build && cd build

# include QT5 in the PATH
PATH="$(brew --prefix qt5)/bin:$PATH" cmake ..
cmake --build .
./tests
curl -O https://zero.sjeng.org/best-network
./leelaz --weights best-network
```

I've added `qt5` to the `brew install` step and used the `brew --prefix` command to ensure that the correct path is used if the user has changed the path which homebrew installs into from the default.

**Comment by wonderingabout** [comment]
except that qt5 is needed only if you want to use autogtp @rakenodiax 

if someone only wants leelaz without wanting to contribute, theres no need to bother him with qt5
i'll try to see how i can implement that in my PR if it can help :+1: 

ty for the feedback

**Comment by wonderingabout** [comment]
here are the mac instructions in my PR btw : 
https://github.com/wonderingabout/leela-zero/tree/minor-clearer-next#example-of-compiling---macos

**Comment by rakenodiax** [comment]
@wonderingabout Thanks! Moving there

**Comment by wonderingabout** [comment]
i'll push a modification, tell me if its ok for you

**Comment by wonderingabout** [comment]
@rakenodiax adding path way sooner or just sooner is the same, but again some mac users will want only `leelaz` and not `autogtp`, since they dont want to contribute no need to bother them with qt5 (and it seems messy on mac)

**Comment by wonderingabout** [comment]
@rakenodiax #2300 addresses this, and i hope fixes it :+1:

**Comment by kellenff** [comment]
@wonderingabout Looks great! Thanks! 👍

**Comment by bcordoba555** [comment]
Ok, it is June 19 2019 and I am trying to install leela-zero 0.17 on macos 10.14.5.  

I followed the instructions in the README files, which do not match the revised instructions shown in this thread BTW.

Homebrew did not install Qt5 so I installed it manually.  (I did not know it was only needed for autogtp, which I did not know what it did until reading this thread).

Anyways, I am not a programmer and I don't know Unix well either so this is hard for this Go player on macos to fix.  

My issue is that Qt5 is not in  /usr/local/Cellar since I manually installed it.  

Here is the error log after I run "cmake .." (inside the build folder as instructed by the README included with 0.17.


--------------------------------

Benoits-MacBook-Pro:build Benoit$ cmake ..
-- Boost version: 1.70.0
-- Found the following Boost libraries:
--   program_options
--   filesystem
-- Using built-in matrix library.
CMake Warning at CMakeLists.txt:50 (find_package):
  By not providing "FindQt5Core.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Qt5Core", but
  CMake did not find one.

  Could not find a package configuration file provided by "Qt5Core" with any
  of the following names:

    Qt5CoreConfig.cmake
    qt5core-config.cmake

  Add the installation prefix of "Qt5Core" to CMAKE_PREFIX_PATH or set
  "Qt5Core_DIR" to a directory containing one of the above files.  If
  "Qt5Core" provides a separate development package or SDK, be sure it has
  been installed.


CMake Warning at CMakeLists.txt:129 (message):
  Qt is not found, build for `autogtp` and `validation` is disabled


-- Configuring done
-- Generating done
-- Build files have been written to: /Users/Benoit/Go/Lizzie/L06/leela-zero/build


---------------------------------------------

Thank you SO MUCH for all the contributors to make this project come to light.  It is exciting!  

I still have not been able to achieve a successful install on macos for two years now).  

Thank you for the help!

Benoît

**Comment by RehmatSChawla** [comment]
@bcordoba555 You can try running `find . -name Qt5CoreConfig.cmake` inside the Qt5Core installation directories to find where the file is and provide that as `Qt5Core_DIR`. I faced a similar issue when trying to use my brew-installed qt@5 to install Geant4:
```
Could not find a package configuration file provided by "Qt5" with any
of the following names:

Qt5Config.cmake
qt5-config.cmake
```
And using the find command, located the first file to be in `/opt/homebrew/Cellar/qt@5/5.15.12/lib/cmake/Qt5`. I provided this as the `Qt5_DIR` and the installation proceeded.

For posterity, I will record my solution to the original problem as well. I too faced the issue of the cmake files for qt5 listing several files that did not exist, including `/opt/homebrew/.//mkspecs/macx-clang` and some other files in `/opt/homebrew/plugins` - a directory which does not exist.

Combining information I found across various issues (including this one), I figured out that there was an error with some environment variable used by the cmake files, which was pointing to `/opt/homebrew/` when it should have been pointing to `/opt/homebrew/Cellar/qt@5/5.15.12/`, where both the `mkspecs` and `plugins` directories were to be found (again, the `find` command helped me track them down). I symlinked the directories to allow the installation to proceed smoothly:
```
$ ln -s /opt/homebrew/Cellar/qt@5/5.15.12/mkspecs /opt/homebrew/mkspecs
$ ln -s /opt/homebrew/Cellar/qt@5/5.15.12/plugins /opt/homebrew/plugins
```
Note that you may need to use `sudo` to execute `ln`.
